xml/tuning-os-tuning.xml

```diff
@@ -0,0 +1,425 @@
+       adding a boot-time kernel parameter as found in the Section 12.4
+       <link xlink:href="https://documentation.suse.com/sles/15-SP1/html/SLES-all/cha-tuning-io.html#cha-tuning-io-barrier"/>.
+       This is not an action that should be taken unilaterally on clusters
+       that contain spinning media devices due to potential performance
+       degradation for those devices. The general result is that there are
+       multiple I/O queues assigned to the device, allowing more jobs to be
+       handled simultaneously by those high performance device such as NVMe
+       and SSD that can service large numbers of requests.
+     </para>
+   </sect2>
+   <sect2>
+     <title>SSD Tuning</title>
+     <para>
+       To get the best read performance, it may be necessary to adjust
+       the <option>read_ahead</option> and write cache settings for the SSD devices.
+       In our particular testing environment, disabling write cache
+       and forcing <option>read_ahead</option> to 2MB resulted in the best
+       overall performance.
+       Before tuning, it is important to check the default values and
+       measure the differences in performance compared to the baseline.
+     </para>
+     <para>
+       By placing the following file in <filename>/etc/udev/rules.d</filename>
+       the device is detected by the model name shown in
+       <filename>/sys/block/{devname}/device/model</filename> and assigned to
+       disable write caching and setting the <option>read_ahead_kb</option> to 2MB.
+     </para>
+<screen>
+  /etc/udev/rules.d/99-ssd.rules
+
+  # Setting specific kernel parameters for a subset of block devices (Intel SSDs)
+  SUBSYSTEM=="block", ATTRS{model}=="INTEL SSDS*", ACTION=="add|change", ATTR{queue/read_ahead_kb}="2048"
+  SUBSYSTEM=="block", ATTRS{model}=="INTEL SSDS*", ACTION=="add|change", RUN+="/sbin/hdparm -W 0 /dev/%k"
+</screen>
+   </sect2>
+   <sect2>
+     <title>Network Stack and Device Tuning</title>
+     <para>
+       Proper tuning of the network stack can substantially add to improving the
+       latency and throughput of the cluster. A full script for the testing
+       we performed can be found in Appendix C.
+     </para>
+     <para>
+       The first and most impactful tuning is to utilize jumbo
+       frame packets. For this to be done, all interfaces utilizing
+       the cluster must be set to the same setting with an MTU of 9000.
+       Network switches often have the ports set to 9100 or higher.
+       This is suitable as they are only passing packets, not creating them.
+     </para>
+     <para>
+       The following salt command ensures the bonded interfaces
+       on all nodes under control (including the test load generation nodes)
+       were utilizing a 9k MTU:
+     </para>
+<screen>
+salt '*' cmd.run 'ip link set bond0 mtu 9000'
+</screen>
+     <para>
+       To set this persistently, utilize YaST to set the MTU for the bonded interface.
```

Comment:
I think we have an entity for YaST although I can't immediately find it.

Reply:
I couldn't either :woman_shrugging: